cmd-koji-upload: Add rpms list in koji manifest

ravanelli opened the pull request:
- Add rpms list from commitmeta.json in the brew manifest output components
- It is part of the effort to improve brew, in order to be compliant with it

Signed-off-by: Renata Ravanelli <rravanel@redhat.com>

ravanelli commented:
I will let it as /hold to validate it with the ProdSec before merging it.

miabbott commented on src/cmd-koji-upload:
```diff
@@ -192,6 +192,13 @@ class Build(_Build):
+        Translate commitmeta.json rpms into a json list
+        Returns the json rpms list
+        """
+        return json.dumps(self.commit["rpmostree.rpmdb.pkglist"])
```
I think we'll need to generate a more structured list according to the example `components` key here - https://docs.pagure.org/koji/content_generator_metadata/#example-metadata-json

The scratch code I worked up:

```python
with open("builds/latest/x86_64/commitmeta.json") as f:
    cm = json.load(f)

components = []

for rpm in cm['rpmostree.rpmdb.pkglist']:
    entry = { "type": "rpm",
        "name": rpm[0],
        "version": rpm[2],
        "release": rpm[3],
        "epoch": rpm[1],
        "arch": rpm[4],
        "sigmd5": None,
        "signature": None }
    components.append(entry)
```

cgwalters reviewed (dismissed):
I don't know the details here, but this looks plausible to me.

miabbott reviewed (dismissed):
LGTM

ravanelli commented:
Umm CI is falling:


`
[2022-04-05T12:57:07.180Z] src/cmd-koji-upload:201:19: E1136: Value 'self.commit' is unsubscriptable (unsubscriptable-object)
[2022-04-05T12:57:09.691Z] make: *** [Makefile:41: pycheck] Error 2`

Any idea how to solve it @miabbott ? Maybe something related to https://github.com/PyCQA/pylint/issues/3139?

miabbott commented:
> Umm CI is falling:
> 
> ` [2022-04-05T12:57:07.180Z] src/cmd-koji-upload:201:19: E1136: Value 'self.commit' is unsubscriptable (unsubscriptable-object) [2022-04-05T12:57:09.691Z] make: *** [Makefile:41: pycheck] Error 2`
> 
> Any idea how to solve it @miabbott ? Maybe something related to [PyCQA/pylint#3139](https://github.com/PyCQA/pylint/issues/3139)?

I can see a couple of `unsubscriptable-object` issues in the PyCQA org - https://github.com/search?q=org%3APyCQA+unsubscriptable-object&type=issues

Since we are able to successfully run this code, I'd say we mask out the check on that line with `# pylint: disable=E1136`

ravanelli commented:
Thanks for the tip Micah!
I did a new build  with some changes, changed the RHEL packages location and added the Fedora (cosa) packages.
Looking better at koji structure seems the Fedora packages should be [here](https://brewweb.engineering.redhat.com/brew/buildrootinfo?buildrootID=8591651),  
and the rhcos packages [here](https://brewweb.engineering.redhat.com/brew/archiveinfo?archiveID=6299577).
However, brew doesn't seem to link the Fedora packages under components as it is done with the RHEL packages. 
I will  add /hold to this PR until I can have an answer for the ProdSec about it.

ravanelli commented:
> Thanks for the tip Micah! I did a new build with some changes, changed the RHEL packages location and added the Fedora (cosa) packages. Looking better at koji structure seems the Fedora packages should be [here](https://brewweb.engineering.redhat.com/brew/buildrootinfo?buildrootID=8591651), and the rhcos packages [here](https://brewweb.engineering.redhat.com/brew/archiveinfo?archiveID=6299577). However, brew doesn't seem to link the Fedora packages under components as it is done with the RHEL packages. I will add /hold to this PR until I can have an answer for the ProdSec about it.

We are still waiting for ProdSec, a feature request was opened in order to allow the metadata link between brew and the manifest, the behavior today only allows packages from RHEL to be linked in brew,  the other distributions as Fedora are ignored. I changed this PR to add into the manifest both packages lists, from cosa (host) and from RHCOs, as described above.

miabbott commented:
> the other distributions as Fedora are ignored.

As long as sticking Fedora RPMs into the manifest doesn't explicitly break anything, this seems fine to me.

miabbott commented on src/cmd-koji-upload:
```diff
@@ -191,6 +191,35 @@ class Build(_Build):
+                "name": name,
+                "version": version,
+                "release": release,
+                "epoch": None,
```
RPMs in RHCOS can have the Epoch set; seems like we should preserve it

travier reviewed (approved):
I don't know this code well but this mostly LGTM
